crates/ide_db/src/helpers/insert_use.rs

```diff
@@ -550,6 +553,15 @@ fn find_insert_position(
             let has_tl = tree.use_tree_list().is_some();
             Some((path, has_tl, node))
         });
+
+    if !group_imports {
+        let nodes: Vec<_> = path_node_iter.collect();
+        if let Some((_, _, node)) = nodes.last().cloned() {
```

Comment:
```suggestion
        if let Some((_, _, node)) = path_node_iter.last() {
```
No need to collect here if I see this right